features/step_definitions/machine.rb

```diff
@@ -62,16 +55,23 @@
   cb["new_node"] = node_name
 end
 
-Then(/^admin ensures node number is restored to #{QUOTED} after scenario$/) do | num_expected |
+Then(/^admin ensures machine number is restored after scenario$/) do
   ensure_admin_tagged
 
   teardown_add {
-    num_actual = 0
-
-    success = wait_for(900, interval: 20) {
-      num_actual = BushSlicer::Node.list(user: admin).length
-      num_expected == num_actual.to_s
+    machines_waiting_delete = BushSlicer::Machine.list(user: admin, project: project("openshift-machine-api")).select { | machine |
+      machine.deleting?
     }
-    raise "Failed to restore cluster, expected number of nodes #{num_expected}, got #{num_actual.to_s}" unless success
+
+    machines_waiting_delete.each do | machine |
+      step %Q{I use the "openshift-machine-api" project}
+      step %Q{I wait for the resource "machine" named "#{machine.name}" to disappear within 1200 seconds}
+      step %Q{the step should succeed}
+    end
+
+    machines = BushSlicer::Machine.list(user: admin, project: project("openshift-machine-api"))
+    unless machines.length == cb.num_to_restore
```

Comment:
This step shouldn't rely on some clipboard value created in another step to operate. In the beginning of step, we should obtain number of machines and use the value here. e.g.
```
machines_orig_num = ...
teardown_add {
  ...
  unless machines.length == machines_orig_num
    ...
}
```
In addition you can store machines into a clipboard for later use in scenario. e.g.
```
cb.all_machines = Machine.list(...)
```

Reply:
Thanks! Now I learned how this works, originally I thought the step executes as last step and assumed machine number obtained there are the current number, thus used the clipboard.
Now understanding it's actually the teardown block gets executed last, this helps me write more decent test, thanks @akostadinov .